testing-unit-common/src/main/java/com/bumble/appyx/testing/unit/common/util/InteropBuilderStub.kt

```diff
@@ -0,0 +1,46 @@
+package com.bumble.appyx.testing.unit.common.util
+
+import androidx.lifecycle.Lifecycle
+import com.bumble.appyx.core.builder.Builder
+import com.bumble.appyx.core.modality.BuildContext
+import com.bumble.appyx.core.node.Node
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class InteropBuilderStub<P>(
+    val delegate: (BuildContext, P) -> Node = { _, _ -> NodeStub() },
+) : Builder<P>() {
+
+    private val _createdNodes = ArrayList<Node>()
+
+    val last: Node?
+        get() = _createdNodes.lastOrNull()
+
+    var lastParam: P? = null
+        private set
+
+    override fun build(buildContext: BuildContext, payload: P): Node =
+        delegate(buildContext, payload).also {
+            _createdNodes += it
+            lastParam = payload
+        }
+
+    fun assertLastParam(param: P) {
+        assertEquals(param, lastParam)
+    }
+
+    fun assertLastParam(assert: P.() -> Boolean) {
+        assertTrue(
```

Comment:
I would split these two assertions into two, otherwise it's tricky to tell what the problem is.

Reply:
I don't think it's necessary to split this 🤔 
If the test fails it will tell you the `lastParams` value in the message, so you will see if it failed because it's null or because it's different from the expected assert

Reply:
the key point is that there is an 'or' involved here which adds ambiguity . Splitting it doesn't cost us much, and can make the users life easier.

They can see that 'oh last param is null' or the param didn't pass the predicate.

Reply:
Done in 9733f65c631ba6a3d30af59207cebeb095f1e872